src/api/index.js

```diff
@@ -0,0 +1,13 @@
+const _getRealms = require('./methods/getRealms')
```

Comment:
Can't this just be another TokenManager? With an additional accompanying method inside the MicrosoftAuthFlow?

Reply:
I was keeping them separate as it wasn't related to auth but we can move it to there if you'd prefer

Reply:
Hmm, I think this belongs more in a separate package than the auth library if it doesn't fit with the current code structure. 

Maybe putting the internals into a minecraftrealms/ folder somewhere where it would easily be movable outside in the future would be a good idea. But ideally if it's placed here for now, the exposed API should conform to the existing API methods and not do something a tangent to the purpose of the lib (authentication). What I mean by that is it would be exposed inside MicrosoftAuthFlow like how we do `getMinecraftBedrockToken()`, `getMinecraftJavaToken()`, etc. Like `getMinecraftRealmToken(options...)`

Reply:
Does that make sense? The relevant options could be passed to the function like the world picker. Maybe a better more descriptive name than “getToken” would suffice. But from the looks of it I do think we should move to a realms package cc @rom1504 

Maybe if you built this out a bit more with this approach it would be clearer the best way to do this

Reply:
Right, so at the moment all this is is a wrapper for the Bedrock Realms API. Authentication is handled by generating an Xbox token with the Realms relying party and then the methods folder will contain different calls. getRealms for example currently returns a list of realms for the authenticating user. Are you under the assumption getRealms is currently returning a token or have I misread it?

Reply:
Yes I understand that. I meant specifically for the documented part of the exposed API. I was looking for something we could easily plug into and use in a single function call inside of the protocol libraries that can do the authentication and return host name/port information, and relevant authentication data (like tokens). If there are other parts of the Realms API that you would implement here that’s OK, but I don’t think we should expose it inside prismarine auth to not do expose things unrelated to basic authentication.  Which is why I said this could be put into a “minecraftrealms” directory somewhere, for future usage inside a separate package. 

That said this may be better to do entirely in a different package that interfaces with prismarine auth. I don’t know the best approach but it sounds like this is the way to go

Reply:
Moved the API to minecraftrealms as suggested and also added the `getMinecraftRealmConnection` this returns ip/port info for the realm that's chosen in the pickRealm function which is passed through